src/rgw/rgw_sal_motr.cc

```diff
@@ -3297,7 +3297,29 @@ int MotrMultipartUpload::list_parts(const DoutPrefixProvider *dpp, CephContext *
   string upload_id = get_upload_id();
 
   if(upload_id.length() == 0){
-    rc = store->get_upload_id(tenant_bkt_name, this->get_key(), upload_id);
+    std::unique_ptr<rgw::sal::Object> obj_ver = this->bucket->get_object(rgw_obj_key(this->get_key()));
```

Comment:
wouldnt it better to keep it in separate function, so we can reuse as needed later also ?

Reply:
Yes, that will help to reuse.
But, afaik, we need to pass the class object here - `this->get_key()`, this might be different for different functions. That can be the target object which has information about the latest object.

Reply:
Ok yes, lets take this optimization separately, better plan more elaborate tests around it, giving enough thoughts on scenarios and then plan to change it